src/core/core.layoutService.js

```diff
@@ -10,24 +10,58 @@ module.exports = function(Chart) {
 	Chart.layoutService = {
 		defaults: {},
 
-		// Register a box to a chartInstance. A box is simply a reference to an object that requires layout. eg. Scales, Legend, Plugins.
+		/**
+		 * Register a box to a chartInstance.
+		 * A box is simply a reference to an object that requires layout. eg. Scales, Legend, Title.
+		 * @method Chart.layoutService.addBox
+		 * @param {Core.Controller} chartInstance the chart to use
+		 * @param {Object} box the box to add
+		 * @param {Number} box.order the order of the boxes. Higher numbers are further away from the chart area
+		 * @param box.options Options for configuring the box
+		 * @param {String} box.options.position the position of the box. Can be 'left', 'top', 'right', 'bottom', or 'chartArea'
+		 * @param {Boolean} box.options.fullWidth if true, and the box is horizontal then push vertical boxes down
+		 * @param {Function} box.isHorizontal return true if this is a horizontal box (top or bottom position)
+		 * @param {Function} box.update Takes two parameters: width and height. Returns size of box
+		 * @param {Function} box.getPadding Returns an object with padding on the edges
+		 * @param {Number} box.width Width of box. Must be valid after update()
+		 * @param {Number} box.height Height of box. Must be valid after update()
+		 * @param {Number} box.left Left edge of the box. Set by layout system and cannot be used in update
+		 * @param {Number} box.top Top edge of the box. Set by layout system and cannot be used in update
+		 * @param {Number} box.right Right edge of the box. Set by layout system and cannot be used in update
+		 * @param {Number} box.bottom Bottom edge of the box. Set by layout system and cannot be used in update
+		 */
 		addBox: function(chartInstance, box) {
 			if (!chartInstance.boxes) {
 				chartInstance.boxes = [];
 			}
+
+			// Ensure that all boxes have an order
+			if (!box.order) {
+				box.order = 0;
```

Comment:
Not sure about this. `sort` is not stable, so I don't think this guarantees the order of boxes. It might be better so simply set `box.order = chartInstance.boxes.length` so that it's more likely to be a unique value.